### PR TITLE
Optimization:Limit Tags Searched For Follow Suggestions

### PR DESCRIPTION
**Pull request by mstruve**
## What type of PR is this? (check all applicable)
- [x] Optimization

## Description
Was recently looking at some slow traces on Honeycomb when I came across one that had a 1 second SQL query in it. Upon further investigation, I noticed that the query was trying to suggest users to follow and was using the current user's followed tags. This particular user had 20 tags they were following which made the query pretty intense with 20 `OR` tagging statements. To save us some process time and power I decided a good solution would be to limit the number of tags when suggesting users to follow. 

## Added tests?
- [x] No, overall behavior has remained the same so current tests should be sufficient 

## Added to documentation?
- [x] No documentation needed

alt_text

### CODE REVIEW

**Review by maestromac — APPROVED** [review]

**Review by juliannatetreault — APPROVED** [review]
Great catch! 👍 👍

**Review by benhalpern — APPROVED** [review]

**Review by rhymes — COMMENTED** [review]
+1 for the idea, is there a chance that sampling will change behavior in a significant way for smaller communities? I'm not 100% sure I follow the "why" behind the logic of `Suggester::Users::Recent` :blush:

**Comment by mstruve** [comment]
>  is there a chance that sampling will change behavior in a significant way for smaller communities (like a few tags and a few users)

If a community has few set suggested tags then this would not affect them at all bc we would likely sample all the tags if the user chose them. The only disadvantage here would be a Forem with a large number of tags and a low number of articles and you happened to select the tags that have no articles. This could happen with the sample or could just happen if the person choose only a couple non popular tags. 

TL;DR I think this change will likely have a very minimal negative effect and preventing large OR queries out weighs that.
